### DISCUSSION TITLE
Is there interest in a bibtex remark plugin?

### DISCUSSION THREAD

**Discussion by songololo** [remarkjs/remark#686] [Ideas] [answered]
Is there interest in a `bibtex` remark plugin for generating bibliographies from citation keys?

This would take a `latex` `bib` file (e.g. from mendeley) and can then generate a references section at the bottom of a `markdown` file based on citations in the markdown text.

For example, if you cite a paper per a citation key in your `bib` file, e.g. `(@Kingma2016)`, it will then add the following at the bottom of your markdown:
> **References**
> Kingma, D. P., Salimans, T., Jozefowicz, R., Chen, X., Sutskever, I., & Welling, M. (2016). Improving Variational Inference with Inverse Autoregressive Flow. http://arxiv.org/abs/1606.04934

> EDIT: I've created a package available at: https://github.com/benchmark-urbanism/remark-bibtex

**Comment by songololo** [comment, marked as answer]
I've created a package here: https://github.com/benchmark-urbanism/remark-bibtex

**Reply by wooorm** [reply]
looks nice! I’m sure this is useful to some folks!

Some ideas:
- you might want to investigate reading the file async (transforms are allowed to be async)
- `fromMarkdown(citedHtml)` sounds weird? That seems HTML
- As you’re doing a lot of things with HTML, working with HTML might be easier than working with markdown: a [rehype](https://github.com/rehypejs/rehype) plugin might make more sense?

**Reply by songololo** [reply]
Thanks, I've revised the package taking your comments into account. I've now switched to using footnotes instead of a randomly added bibliography section.
https://github.com/benchmark-urbanism/remark-bibtex
